README.md

```diff
@@ -30,6 +30,251 @@ you're a tool author you may be interested in the formal [EBNF grammar][].
+`fluent.syntax`.
+
+Usage
+-----
+
+To generate translations using the ``fluent.runtime`` package, you start with
+the `FluentBundle` class:
+
+    >>> from fluent.bundle import FluentBundle
+
+You pass a list of locales to the constructor - the first being the desired
+locale, with fallbacks after that:
+
+    >>> bundle = FluentBundle(["en-US"])
+
+
+You must then add messages. These would normally come from a `.ftl` file stored
+on disk, here we will just add them directly:
+
+    >>> bundle.add_messages("""
+    ... welcome = Welcome to this great app!
+    ... greet-by-name = Hello, { $name }!
+    ... """)
+
+To generate translations, use the `format` method, passing a message ID and an
+optional dictionary of substitution parameters. If the the message ID is not
+found, a `LookupError` is raised. Otherwise, as per the Fluent philosophy, the
+implementation tries hard to recover from any formatting errors and generate the
+most human readable representation of the value. The `format` method therefore
+returns a tuple containing `(translated string, errors)`, as below.
+
+    >>> translated, errs = bundle.format('welcome')
+    >>> translated
+    "Welcome to this great app!"
+    >>> errs
+    []
+
+    >>> translated, errs = bundle.format('greet-by-name', {'name': 'Jane'})
+    >>> translated
+    'Hello, \u2068Jane\u2069!'
+
+    >>> translated, errs = bundle.format('greet-by-name', {})
+    >>> translated
+    'Hello, \u2068name\u2069!'
+    >>> errs
+    [FluentReferenceError('Unknown external: name')]
+
+You will notice the extra characters `\u2068` and `\u2069` in the output. These
+are Unicode bidi isolation characters that help to ensure that the interpolated
+strings are handled correctly in the situation where the text direction of the
+substitution might not match the text direction of the localized text. These
+characters can be disabled if you are sure that is not possible for your app by
+passing `use_isolating=False` to the `FluentBundle` constructor.
+
+Python 2
+--------
+
+The above examples assume Python 3. Since Fluent uses unicode everywhere
+internally (and doesn't accept bytestrings), if you are using Python 2 you will
+need to make adjustments to the above example code. Either add `u` unicode
+literal markers to strings or add this at the top of the module or the start of
+your repl session:
+
+    from __future__ import unicode_literals
+
+
+Numbers
+-------
+
+When rendering translations, Fluent passes any numeric arguments (int or float)
+through locale-aware formatting functions:
+
+    >>> bundle.add_messages("show-total-points = You have { $points } points.")
+    >>> val, errs = bundle.format("show-total-points", {'points': 1234567})
+    >>> val
+    'You have 1,234,567 points.'
+
+
+You can specify your own formatting options on the arguments passed in by
+wrapping your numeric arguments with `fluent.types.fluent_number`:
+
+    >>> from fluent.bundle.types import fluent_number
```

Comment:
And here: `from fluent.runtime.types`.

Reply:
Similarly here. In fact, if we wanted, we could also promote some of these user-facing things into higher level - e.g. `fluent.bundle.types` could go to `fluent.types`, which might make importing `fluent_number` and `fluent_date` a bit handier. We'd just have to add `fluent.types` to setup.py to enable this.

Reply:
I think you're right, we'd need that too. Your other comment convinced be that it is probably better to go with `fluent.runtime`, it would be surprising to not have `fluent.runtime` importable.